substrate-node/tests/TfChainClient.py

```diff
@@ -0,0 +1,841 @@
+from cmath import exp
+from datetime import datetime
+from re import S
+from unittest.mock import DEFAULT
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
+from itertools import count
+import json
+import logging
+from random import randbytes
+import time
+from substrateinterface import SubstrateInterface, Keypair
+
+GIGABYTE = 1024*1024*1024
+
+TIMEOUT_WAIT_FOR_BLOCK = 6
+
+DEFAULT_SIGNER = "Alice"
+DEFAULT_PORT = 9945
+
+FARM_CERTIFICATION_NOTCERTIFIED = "NotCertified"
+FARM_CERTIFICATION_GOLD = "Gold"
+FARM_CERTIFICATION_TYPES = [
+    FARM_CERTIFICATION_NOTCERTIFIED, FARM_CERTIFICATION_GOLD]
+
+NODE_CERTIFICATION_DIY = "Diy"
+NODE_CERTIFICATION_CERTIFIED = "Certified"
+NODE_CERTIFICATION_TYPES = [
+    NODE_CERTIFICATION_DIY, NODE_CERTIFICATION_CERTIFIED]
+
+UNIT_BYTES = "Bytes"
+UNIT_KILOBYTES = "Kilobytes"
+UNIT_MEGABYTES = "Mebabytes"
+UNIT_GIGABYTES = "Gigabytes"
+UNIT_TERRABYTES = "Terrabytes"
+UNIT_TYPES = [UNIT_BYTES, UNIT_KILOBYTES,
+              UNIT_MEGABYTES, UNIT_GIGABYTES, UNIT_TERRABYTES]
+
+
+class TfChainClient:
+    def __init__(self):
+        self._wait_for_finalization = False
+        self._wait_for_inclusion = True
+        self._create_keys()
+
+    def _connect_to_server(self, url: str):
+        return SubstrateInterface(url=url, ss58_format=42, type_registry_preset='polkadot')
+
+    def _create_keys(self):
+        self._predefined_keys = {}
+        self._predefined_keys["Alice"] = Keypair.create_from_uri("//Alice")
+        self._predefined_keys["Bob"] = Keypair.create_from_uri("//Bob")
+        self._predefined_keys["Charlie"] = Keypair.create_from_uri(
+            "//Charlie")
+        self._predefined_keys["Dave"] = Keypair.create_from_uri(
+            "//Dave")
+        self._predefined_keys["Eve"] = Keypair.create_from_uri("//Eve")
+        self._predefined_keys["Ferdie"] = Keypair.create_from_uri("//Ferdie")
+
+    def _setup_predefined_account(self, name: str, port: int = DEFAULT_PORT, create_twin: bool = False):
```

Comment:
would give this function a better name, something like `setup_offchain_workers` ? Since this function handles the setup for the offchain workers